### PR TITLE
Add type annotation on `concurrency/solo.py`

### PR DESCRIPTION
**Pull request by Kludex**
- Related to #7394

### CODE REVIEW

File: celery/concurrency/solo.py [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
-            'max-concurrency': 1,
-            'processes': [os.getpid()],
-            'max-tasks-per-child': None,
-            'put-guarded-by-semaphore': True,
-            'timeouts': (),
+            "max-concurrency": 1,
+            "processes": [os.getpid()],
+            "max-tasks-per-child": None,
+            "put-guarded-by-semaphore": True,
+            "timeouts": (),
```
Comment (Kludex) [inline celery/concurrency/solo.py:46]: Side effect from my IDE. If it bothers, accept the following suggestion:
```suggestion
            'max-concurrency': 1,
            'processes': [os.getpid()],
            'max-tasks-per-child': None,
            'put-guarded-by-semaphore': True,
            'timeouts': (),
```

**Review by thedrow — APPROVED** [review]

**Comment by thedrow** [comment]
Can you please fix the merge conflict?

**Comment by auvipy** [comment]
I am not sure why the CI is failing

**Comment by Nusnus** [comment]
> I am not sure why the CI is failing

This error:
```
2023-01-29T15:44:18.2082260Z >       sqlite = dialect.dbapi()
```

It is fixed by the merged PR: #8033

This PR needs to be rebased on master @Kludex

**Pull request closed without merging**